app/src/main/java/cm/aptoide/pt/DeepLinkIntentReceiver.java

```diff
@@ -98,290 +97,347 @@
+    long id = Long.parseLong(strings[strings.length - 1].split("\\.myapp")[0]);
+    return startFromAppView(id, null, false);
+  }
 
-          Logger.d("AptoideAptWord", json);
+  private Intent dealWithWebservicesAptoide(String uri) {
+    /** refactored to remove org.apache libs */
+    Map<String, String> params = null;
 
-          GetAdsResponse.Ad ad = null;
-          try {
-            ad = new ObjectMapper().readValue(json, GetAdsResponse.Ad.class);
-          } catch (IOException e) {
-            CrashReport.getInstance()
-                .log(e);
-          }
+    try {
+      params = AptoideUtils.StringU.splitQuery(URI.create(uri));
+    } catch (UnsupportedEncodingException e) {
+      CrashReport.getInstance()
+          .log(e);
+    }
 
-          if (ad != null) {
-            Intent i = new Intent(this, startClass);
-            i.putExtra(DeepLinksTargets.FROM_AD, adMapper.map(ad));
-            startActivity(i);
-          } else {
-            deepLinkAnalytics.sendWebsite();
-            finish();
-          }
+    if (params != null) {
+      String uid = null;
+      for (Map.Entry<String, String> entry : params.entrySet()) {
+        if (entry.getKey()
+            .equals("uid")) {
+          uid = entry.getValue();
         }
-      } else if (uri.startsWith("file://")) {
+      }
 
-        downloadMyApp();
-      } else if (uri.startsWith("aptoideinstall://")) {
-        parseAptoideInstallUri(uri.substring("aptoideinstall://".length()));
-      } else if (u.getHost()
-          .equals("cm.aptoide.pt") && u.getPath()
-          .equals("/deeplink") && u.getQueryParameter("name")
-          .equals("getHome")) {
-        String id = u.getQueryParameter("user_id");
-        if (id != null) {
-          openUserScreen(Long.valueOf(id));
-        }
-        deepLinkAnalytics.sendWebsite();
-        finish();
-        return;
-      } else if (uri.startsWith("aptoide://")) {
-        Uri parse = Uri.parse(uri);
-        if ("getUserTimeline".equals(parse.getQueryParameter("name"))) {
-          String cardId = parse.getQueryParameter("cardId");
-          startFromAppsTimeline(cardId);
-          deepLinkAnalytics.sendWebsite();
-          finish();
-          return;
+      if (uid != null) {
+        try {
+          long id = Long.parseLong(uid);
+          return startFromAppView(id, null, true);
+        } catch (NumberFormatException e) {
+          CrashReport.getInstance()
+              .log(e);
+          CrashReport.getInstance()
+              .log(e);
+          ShowMessage.asToast(getApplicationContext(), R.string.simple_error_occured + uid);
         }
-        if ("search".equals(parse.getQueryParameter("name"))) {
-          String query = "";
-          if (parse.getQueryParameterNames()
-              .contains("keyword")) {
-            query = parse.getQueryParameter("keyword");
+      }
+    }
+    return null;
+  }
+
+  private Intent dealWithShortcuts() {
```

Comment:
Why does this return an intent? Seems like it could/should be void

Reply:
done